doc/code_checker.rst

```diff
@@ -0,0 +1,28 @@
+# Static Code analysis with CodeChecker
+
+Wakaama uses [`CodeChecker`](https://codechecker.readthedocs.io/) for static code analysis.
+
+It's possible to run `CodeChecker` in two different modes: `full` and `diff`
+
+In `full` mode all found issues are reported. In `diff` mode only new issues are shown.
+
+The `diff` mode compares found issues with a 'base line' and shows only newly found issues.
+
+## Running `CodeChecker`
+
+The `CodeChecker` is run as part of the CI GitHub Actions. But it can be run manually:
+
+To show new issues:
+```
+tools/ci/run_ci.sh --run-build --run-code-checker --code-checker diff
+```
+
+To show *all* issues:
+```
+tools/ci/run_ci.sh --run-build --run-code-checker --code-checker full
```

Comment:
Does not work for me:
```
<snip>
[INFO 2024-12-04 18:29] - No checkers enabled for gcc
[ERROR 2024-12-04 18:29] - No checker(s) with these names was found:
cplusplus
<snip>
```

Reply:
Strange... Did you install `clang-tools-18` and `cppcheck`?

Reply:
Thanks! `clang-tools-18` was missing 🙈